File: lib/src/model/FortuneUser/components/user_settings.dart
import 'package:dg_fortune/src/enums/theme_type_enum.dart';

import 'notification_settings.dart';

class UserSettings {
  NotificationSettings notificationSettings;
  ThemeType themeType;

  UserSettings({
    required this.notificationSettings,
    required this.themeType,
  });

  factory UserSettings.fromJson(Map<String, dynamic> json) => UserSettings(
        notificationSettings: NotificationSettings.fromJson(json["notificationSettings"]),
        themeType: themeTypeEnumFromString(json["themeType"] as String),
      );

  Map<String, dynamic> toJson() => {
        "notificationSettings": notificationSettings.toJson(),
        "themeType": themeType.name.toString(),
      };
}

File: functions/model/FortuneUser/components/UserSettings.js
const { NotificationSettings } = require('./NotificationSettings');



class UserSettings {
    constructor({ notificationSettings, themeType }) {
        this.notificationSettings = notificationSettings;
        this.themeType = themeType;


    }

    static fromJson(json) {
        return new UserSettings({
            notificationSettings: NotificationSettings.fromJson(json["notificationSettings"]),
            themeType: json["themeType"]
        });
    }

    toJson() {
        return {
            "notificationSettings": this.notificationSettings.toJson(),
            "themeType": this.themeType,
        };
    }
}
module.exports = { UserSettings };


